packages/amplify-category-api/src/provider-utils/awscloudformation/utils/rds-resources/utils.ts

```diff
@@ -50,6 +54,13 @@ export const checkForUnsupportedDirectives = (schema: string, context: DataSourc
   visit(document, schemaVisitor);
 };
 
+export const containsSqlModelOrDirective = (dataSourceStrategies: Record<string, ModelDataSourceStrategy>): boolean => {
+  if (dataSourceStrategies && Object.keys(dataSourceStrategies).length > 0) {
```

Comment:
`dataSourceStrategies` is guaranteed to be non-null; you can simplify this by simply returning the result of the `Object.values().some` call below.

If we're worried that `dataSourceStrategies` might be null, we need to explore the call path carefully. We intentionally built in checks that strategies must always be present in the transformer internals, and if we're violating that invariant we need to find out about it.

Reply:
Makes sense. I will update as suggested.